internal/apps_registry/cached_app_registry.go

```diff
@@ -171,12 +171,22 @@ func arePoktApplicationSignersEqual(slice1, slice2 []*models.PoktApplicationSign
 		return sortedSlice2[i].NetworkApp.Address < sortedSlice2[j].NetworkApp.Address
 	})
 
-	// Now that slices are sorted, check if address keys are same.
+	// Now that slices are sorted, check if address keys and chains are same.
 	for i := range slice1 {
-		// Check if any field is different
-		if !strings.EqualFold(sortedSlice1[i].NetworkApp.Address, sortedSlice2[i].NetworkApp.Address) {
+
+		networkApp1 := sortedSlice1[i].NetworkApp
+		networkApp2 := sortedSlice2[i].NetworkApp
+
+		// If address are not same, gateway operator added or remove some application
+		if !strings.EqualFold(networkApp1.Address, networkApp2.Address) {
+			return false
+		}
+
+		// Gateway operator may have updated chains staked
+		if !strings.EqualFold(strings.Join(networkApp1.Chains, ","), strings.Join(networkApp2.Chains, ",")) {
```

Comment:
@nodiesBlade what if the order of Chain IDs is different for two applications?
Let's say:
- networkApp1.Chains = ["111", "222", "333"]
- networkApp2.Chains = ["222", "333", "111"]

I don't know if `Chains` are ordered before assigning to `NetworkApp`, but it is better not to rely on string concatenation equality.

Reply:
+ worth adding a unit test for such a case.

Reply:
Due to the nature of blockchain and how hashing works, if the application ordering change, i would argue the application is still technically different and we should recache it. and so that's why I took a little shortcut. 

Though I'm glad you mentioned it as well, wouldn't mind adding the extra security.

Reply:
Yeah, I was talking only about the way Chains were compared. 

Absolutely agree with what you say about Applications equivalency on different Chains order.